2005-03-02  Daniel Jacobowitz  <dan@codesourcery.com>

	* linux-nat.c (linux_supports_tracevforkdone, child_follow_fork):
	Use LWP IDs.

--- gdb-6.3/gdb/linux-nat.c.fix	2005-03-03 17:26:33.000000000 -0500
+++ gdb-6.3/gdb/linux-nat.c	2005-03-03 17:29:13.000000000 -0500
@@ -287,9 +287,12 @@ linux_supports_tracevforkdone (int pid)
 void
 linux_enable_event_reporting (ptid_t ptid)
 {
-  int pid = ptid_get_pid (ptid);
+  int pid = ptid_get_lwp (ptid);
   int options;
 
+  if (pid == 0)
+    pid = ptid_get_pid (ptid);
+
   if (! linux_supports_tracefork (pid))
     return;
 
@@ -334,7 +337,9 @@ child_follow_fork (int follow_child)
 
   get_last_target_status (&last_ptid, &last_status);
   has_vforked = (last_status.kind == TARGET_WAITKIND_VFORKED);
-  parent_pid = ptid_get_pid (last_ptid);
+  parent_pid = ptid_get_lwp (last_ptid);
+  if (parent_pid == 0)
+    parent_pid = ptid_get_pid (last_ptid);
   child_pid = last_status.value.related_pid;
 
   if (! follow_child)
